src/Parasites/Genotype.cpp

```diff
@@ -196,49 +196,49 @@ void Genotype::calculate_EC50_power_n(const PfGeneInfo &gene_info, DrugDatabase
   }
 }
 
-Genotype *Genotype::perform_mutation_by_drug(Config *pConfig, Random *pRandom, DrugType *pDrugType) const {
+Genotype *Genotype::perform_mutation_by_drug(Config *pConfig, Random *pRandom, DrugType *pDrugType, double mutation_probability_by_locus) const {
   std::string new_aa_sequence { aa_sequence };
-
-  // draw random aa position
-  auto aa_pos_id = pRandom->random_uniform(pDrugType->resistant_aa_locations.size());
-  auto aa_pos = pDrugType->resistant_aa_locations[aa_pos_id];
-
-  if (pConfig->mutation_mask()[aa_pos.aa_index_in_aa_string] == '1') {
-    if (aa_pos.is_copy_number) {
-      // increase or decrease by 1 step
-      auto old_copy_number = NumberHelpers::char_to_single_digit_number(aa_sequence[aa_pos.aa_index_in_aa_string]);
-      if (old_copy_number == 1) {
-        new_aa_sequence[aa_pos.aa_index_in_aa_string] = NumberHelpers::single_digit_number_to_char(old_copy_number + 1);
-      } else if (old_copy_number
-                 == pConfig->pf_genotype_info()
+  for(int aa_pos_id = 0; aa_pos_id < pDrugType->resistant_aa_locations.size(); aa_pos_id++) {
+    // get aa position info (aa index in aa string, is copy number)
+    auto aa_pos = pDrugType->resistant_aa_locations[aa_pos_id];
+    if(pConfig->mutation_mask()[aa_pos.aa_index_in_aa_string] == '1'){
```

Comment:
Why not invert this statement and reduce some of the nesting of the if statements below?

Reply:
We only allow mutation happen if the mask is on.